src/variable_info.hpp

```diff
@@ -16,98 +16,119 @@
 
 #pragma once
 
-/** Information on a WML variable. */
-#include <string>
 #include "config.hpp"
 #include "variable_info_detail.hpp"
 
-class invalid_variablename_exception : public std::exception
+#include <string>
+#include <type_traits>
+
+// Restrict alias to local scope.
+namespace
 {
-public:
-	invalid_variablename_exception() : std::exception() {}
-	const char* what() const NOEXCEPT
-	{
-		return "invalid_variablename_exception";
-	}
-};
+namespace impl = variable_info_implementation;
+}
 
-template<const variable_info_detail::variable_info_type vit>
+/** Information on a WML variable. */
+template<typename vit>
```

Comment:
I still don't understand why you've changed this from a non-type to a type parameter.